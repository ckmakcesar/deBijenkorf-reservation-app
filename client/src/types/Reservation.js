import * as PropTypes from 'prop-types';

const Reservation = {
  id: PropTypes.number.isRequired,
  storeId: PropTypes.number.isRequired,
  statusId: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
};

export default Reservation;
