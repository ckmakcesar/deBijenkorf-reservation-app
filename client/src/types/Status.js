import * as PropTypes from 'prop-types';

const Status = {
  id: PropTypes.number.isRequired,
  code: PropTypes.string.isRequired,
};

export default Status;
