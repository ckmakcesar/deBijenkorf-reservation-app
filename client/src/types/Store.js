import * as PropTypes from 'prop-types';

const Store = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
};

export default Store;
