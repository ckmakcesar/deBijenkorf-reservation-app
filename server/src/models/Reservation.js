const { Sequelize, DataTypes } = require('sequelize');

const { sequelize } = require('.');

const Reservation = sequelize.define('reservation', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  name: { type: DataTypes.STRING, allowNull: false },
  storeId: { type: DataTypes.INTEGER, allowNull: false },
  statusId: { type: DataTypes.INTEGER, allowNull: false },
  time: { type: DataTypes.DATEONLY, allowNull: false },
  createdAt: { type: DataTypes.DATE, allowNull: false, defaultValue: Sequelize.NOW },
}, {
  freezeTableName: true,
  underscored: true,
  version: false,
  updatedAt: false,
});

module.exports = Reservation;
