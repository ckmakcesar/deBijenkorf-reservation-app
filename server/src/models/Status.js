const { DataTypes } = require('sequelize');

const { sequelize } = require('.');

const Status = sequelize.define('status', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  code: { type: DataTypes.STRING, allowNull: false, unique: true },
  name: { type: DataTypes.STRING, allowNull: false, unique: true },
}, {
  freezeTableName: true,
  underscored: true,
  version: false,
  createdAt: false,
  updatedAt: false,
});

module.exports = Status;
