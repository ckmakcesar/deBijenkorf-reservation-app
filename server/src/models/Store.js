const { DataTypes } = require('sequelize');

const { sequelize } = require('.');
const Reservation = require('./Reservation');

const Store = sequelize.define('status', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  name: { type: DataTypes.STRING, allowNull: false, unique: true },
  address: { type: DataTypes.STRING, allowNull: false },
}, {
  freezeTableName: true,
  underscored: true,
  version: false,
  createdAt: false,
  updatedAt: false,
});

// Associations
Store.hasMany(Reservation, {
  sourceKey: 'id',
  foreignKey: 'storeId',
  as: 'reservations',
})
Reservation.belongTo(Store, {
  foreignKey: 'storeId',
  as: 'store',
})

module.exports = Store;
