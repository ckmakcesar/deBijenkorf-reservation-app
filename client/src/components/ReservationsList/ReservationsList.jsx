import React from 'react';
import PropTypes from 'prop-types';

import ReservationDetails from '../ReservationDetails';
import Reservation from '../../types/Reservation';
import Store from '../../types/Store';
import Status from '../../types/Status';

import styles from '../../styles/ReservationsList.module.css';

const ReservationsList = ({
  reservationsMap,
  loading,
  storesMap,
  statusesMap,
  setReservationToDelete,
  setDrawerReservationId,
}) => {
  if (loading) {
    return (<div>LOADING DISC</div>); // TODO
  }

  const reservationCount = Object.keys(reservationsMap).length;

  return (
    <>
      {!reservationCount
        ? <div className={styles.emptyState}>There are no reservations</div> // TODO - EMPTY STATE
        : Object.values(reservationsMap).map((reservation) => (
          <div key={reservation.id}>
            <ReservationDetails
              reservation={reservation}
              store={storesMap[reservation.storeId]}
              status={statusesMap[reservation.statusId]}
              setReservationToDelete={setReservationToDelete}
              setDrawerReservationId={setDrawerReservationId}
            />
            <hr id='dotted' />
          </div>
        ))
      }
    </>
  );
};

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
ReservationsList.propTypes = {
  reservationsMap: PropTypes.objectOf(PropTypes.exact(Reservation)).isRequired,
  loading: PropTypes.bool.isRequired,
  storesMap: PropTypes.objectOf(PropTypes.exact(Store)).isRequired,
  statusesMap: PropTypes.objectOf(PropTypes.exact(Status)).isRequired,
  setReservationToDelete: PropTypes.func.isRequired,
  setDrawerReservationId: PropTypes.func.isRequired,
};

export default ReservationsList;
