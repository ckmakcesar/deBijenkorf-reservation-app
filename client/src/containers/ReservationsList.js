import { connect } from 'react-redux';

import { reservationsMapSelector, reservationsLoadingSelector } from '../selectors/reservations';
import ReservationsList from '../components/ReservationsList';

const mapStateToProps = (state) => ({
  reservationsMap: reservationsMapSelector(state),
  loading: reservationsLoadingSelector(state),
});

export default connect(mapStateToProps)(ReservationsList);
