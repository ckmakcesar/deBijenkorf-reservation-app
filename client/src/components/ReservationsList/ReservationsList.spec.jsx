import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ReservationsList from './ReservationsList';
import { mockReservationsMap } from '../../tests/data/Reservation';
import { mockStoresMap } from '../../tests/data/Store';
import { mockStatusesMap } from '../../tests/data/Status';

describe('component:ReservationsList', () => {
  it('Should render ReservationsList with every optional props present', () => {
    const wrapper = shallow(
      <ReservationsList
        reservationsMap={mockReservationsMap}
        loading={false}
        storesMap={mockStoresMap}
        statusesMap={mockStatusesMap}
        setReservationToDelete={jest.fn()}
        setDrawerReservationId={jest.fn()}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
