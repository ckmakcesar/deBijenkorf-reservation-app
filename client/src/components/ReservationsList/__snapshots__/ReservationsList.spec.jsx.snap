// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:ReservationsList Should render ReservationsList with every optional props present 1`] = `
<div
  id="Reservations-list"
>
  <div
    id="Reservation-details-1"
    key="1"
  >
    <ReservationDetails
      reservation={
        Object {
          "createdAt": "2022-02-10T23:00:00.000Z",
          "date": "2022-01-26T23:00:00.000Z",
          "id": 1,
          "name": "Repair",
          "statusId": 3,
          "storeId": 1,
        }
      }
      setDrawerReservationId={[MockFunction]}
      setReservationToDelete={[MockFunction]}
      status={
        Object {
          "code": "TODO",
          "id": 3,
          "name": "Todo",
        }
      }
      store={
        Object {
          "address": "Mid Amsterdam",
          "id": 1,
          "name": "Centraal",
        }
      }
    />
    <hr
      className="dotted"
    />
  </div>
  <div
    id="Reservation-details-2"
    key="2"
  >
    <ReservationDetails
      reservation={
        Object {
          "createdAt": "2022-02-20T23:00:00.000Z",
          "date": "2022-01-26T23:00:00.000Z",
          "id": 2,
          "name": "Consultation",
          "statusId": 3,
          "storeId": 1,
        }
      }
      setDrawerReservationId={[MockFunction]}
      setReservationToDelete={[MockFunction]}
      status={
        Object {
          "code": "TODO",
          "id": 3,
          "name": "Todo",
        }
      }
      store={
        Object {
          "address": "Mid Amsterdam",
          "id": 1,
          "name": "Centraal",
        }
      }
    />
    <hr
      className="dotted"
    />
  </div>
  <div
    id="Reservation-details-3"
    key="3"
  >
    <ReservationDetails
      reservation={
        Object {
          "createdAt": "2022-02-28T23:00:00.000Z",
          "date": "2022-01-26T23:00:00.000Z",
          "id": 3,
          "name": "Gift Drop",
          "statusId": 2,
          "storeId": 1,
        }
      }
      setDrawerReservationId={[MockFunction]}
      setReservationToDelete={[MockFunction]}
      status={
        Object {
          "code": "IN_PROGRESS",
          "id": 2,
          "name": "In Progress",
        }
      }
      store={
        Object {
          "address": "Mid Amsterdam",
          "id": 1,
          "name": "Centraal",
        }
      }
    />
    <hr
      className="dotted"
    />
  </div>
</div>
`;
